src/test/librbd/fsx.cc

```diff
@@ -1183,7 +1185,22 @@ krbd_resize(struct rbd_ctx *ctx, uint64_t size)
 	if (ret < 0)
 		return ret;
 
-	return __librbd_resize(ctx, size);
+	ret = __librbd_resize(ctx, size);
+	if (ret < 0)
+		return ret;
+	do {
+		sleep(count++);
+		ret = krbd_get_size(ctx, &effective_size);
+		if (ret < 0) {
+			prt("krbd_get_size failed\n");
+			return ret;
+		}
+		prt("Sizes: expected (0x%llx), effective (0x%llx)\n",
+		    (unsigned long long)size,
+		    (unsigned long long)effective_size);
+	} while (size != effective_size);
```

Comment:
The retry loop should be bounded.  I would suggest something like `for (int delay = 250; delay < 4000; delay += 250)` and  `usleep(delay * 1000)` for an upper bound of 15 retries and 30 seconds of cumulative sleep.

Reply:
we can use the `count` to limit the loop to max 5 or 10 times if you want it to be bounded.